python_modules/dagster/dagster/_core/storage/captured_log_manager.py

```diff
@@ -0,0 +1,217 @@
+# pylint: disable=unused-argument
+
+from abc import ABC, abstractmethod
+from contextlib import contextmanager
+from typing import List, NamedTuple, Optional
+
+MAX_BYTES_CHUNK_READ = 4194304  # 4 MB
+
+
+class CapturedLogData(
+    NamedTuple(
+        "_CapturedLogData",
+        [
+            ("log_key", List[str]),
+            ("stdout", Optional[bytes]),
+            ("stderr", Optional[bytes]),
+            ("cursor", Optional[str]),
+        ],
+    )
+):
+    """
+    Object representing captured log data, either a partial chunk of the log data or the full
+    capture.  Contains the raw bytes and optionally the cursor offset for the partial chunk.
+    """
+
+    def __new__(
+        cls,
+        log_key: List[str],
+        stdout: Optional[bytes] = None,
+        stderr: Optional[bytes] = None,
+        cursor: Optional[str] = None,
+    ):
+        return super(CapturedLogData, cls).__new__(cls, log_key, stdout, stderr, cursor)
```

Comment:
having both out and err streams in the same payload seems like it might be awkward. I didn't think much of it until i saw the combined cursor.

what led you to this set-up?

Reply:
I first had them split out, but it just seemed easier to group them together.  I don't have super strong feelings here...  The frontend subscribes/unsubscribes to both at the same time, each with its own subscription.